src/main/java/org/dmfs/jems/single/elementary/Digested.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.jems.single.elementary;
+
+import org.dmfs.iterables.decorators.Mapped;
+import org.dmfs.iterables.elementary.Seq;
+import org.dmfs.iterators.Function;
+import org.dmfs.jems.function.BiFunction;
+import org.dmfs.jems.function.elementary.SingleFunction;
+import org.dmfs.jems.messagedigest.MessageDigestFactory;
+import org.dmfs.jems.single.Single;
+
+import java.io.UnsupportedEncodingException;
+import java.security.MessageDigest;
+import java.util.Locale;
+
+
+/**
+ * A {@link Single} of a byte array which represents the digested value of the given input data.
+ *
+ * @author Marten Gajda
+ */
+public final class Digested implements Single<byte[]>
```

Comment:
Since we call the result 'digest' as well, could it be just `Digest`?

Reply:
yep, makes sense.